tests/test_material.py

```diff
@@ -23,328 +23,294 @@
-            # Try getting a value from the correlation; should be
-            # a float, and should be greater than 0
-            assert getattr(mat, prop) > 0.0
-
-
-def test_error_table_non_positive_val(testdir, caplog):
-    """Test error when table has negative value"""
-    f = os.path.join(testdir, 'test_inputs', 'custom_mat-3.csv')
-    # with pytest.raises(SystemExit):
-    #     Material('badbad', from_file=f)
-    with pytest.raises(SystemExit):
-        Material('badbad', from_file=f)     
-    assert 'Non-positive or missing values detected in material data ' in caplog.text
-
-
-def test_material_coeff_from_file(testdir):
-    """Try loading material property correlation coeffs from CSV"""
-    filepath = os.path.join(testdir, 'test_inputs', 'custom_mat.csv')
-    m = Material('sodium', from_file=filepath)
-    m.update(623.15)
-    for prop in ['thermal_conductivity', 'heat_capacity',
-                 'density', 'viscosity']:
-        assert hasattr(m, prop)
-        assert getattr(m, prop) > 0.0
-
-
-def test_failed_material(caplog):
-    """Make sure that the Material class fails with bad input"""
-    with pytest.raises(SystemExit):
-        Material('candycorn')
-    assert 'material candycorn' in caplog.text
-
-
-def test_bad_temperature(caplog):
-    """Make sure Material throws error for negative temperatures"""
-    m = Material('sodium')
-    with pytest.raises(SystemExit):
-        m.update(0.0)
-    assert 'must be > 0; given' in caplog.text
-
-
-def test_bad_property(caplog):
-    """Make sure Material throws error for negative temperatures"""
-    # Define a custom dictionary and use it
-    c = {'thermal_conductivity': [0.05, 1.0, 2.0],
-         'heat_capacity': [480.0, 0.5],
-         'density': [1200.0, 0.2, 0.03],
-         'viscosity': [300.0, -1.0],
-         }
-    m = Material('test', coeff_dict=c)
-    with pytest.raises(SystemExit):
-        m.update(400.0)
-    assert 'viscosity must be > 0; given' in caplog.text
-
-
-def test_sodium_interpolated_value():
-    """Test that DASSH properly interpolates sodium properties"""
-    rho_400 = 919.0
-    rho_500 = 897.0
-    ans = np.average([rho_400, rho_500])
-    sodium = Material('sodium', temperature=450.0)
-    print('ans =', ans)
-    print('res =', sodium.density)
-    err = (sodium.density - ans) / ans
-    print('err = ', err)
-    assert ans == pytest.approx(sodium.density)
-
-
-def test_table_with_missing_values(testdir, caplog):
-    """Test that DASSH interpolates properties
-    with missing or zero values"""
-    filepath = os.path.join(testdir, 'test_inputs', 'custom_mat-2.csv')
-    with pytest.raises(SystemExit):
-        Material('missing_sodium', from_file=filepath)    
-    assert 'Non-positive or missing values detected in material data ' in caplog.text
-    
-def __test_property(mat, t_range, correct_values, property):
+def property_test(mat: str, t_range: np.array, correct_values: list[float], property: str):
```

Comment:
Maybe to modify "t_range" name into "t_values" to avoid any misunderstanding: this variable represents the array of temperature values the properties are evaluated at.